libraries/standard/mqtt/src/mqtt.c

```diff
@@ -323,6 +365,15 @@ static int32_t recvExact( const MQTTContext_t * pContext,
     pIndex = pContext->networkBuffer.pBuffer;
     recvFunc = pContext->transportInterface.recv;
     getTimeStampMs = pContext->callbacks.getTime;
+
+    /* If no time function is provided, use a function that always returns 0 so
+     * as to run the loop for a single iteration. */
+    if( getTimeStampMs == NULL )
+    {
+        getTimeStampMs = getTimeStampNoSystemTime;
+        assert( timeoutMs == 0U );
+    }
```

Comment:
This strikes me as odd.  Are we trying to support systems without clocks?  If so, is a dummy function better than another conditional on timeoutMs?

Reply:
Yes, the clock is not required. I added the dummy function since I did not want to increase the complexity of the code too much by adding conditionals for the time function. After testing it however, the complexity is still within our limits despite its increase, so I pushed the change